book/scheduling-and-threading.md

```diff
@@ -1215,91 +1149,119 @@ class Tab:
 
     def run_animation_frame(self):
         self.render()
+        commit = CommitForRaster(
+            url=self.url,
+            scroll=self.scroll,
+            height=document_height,
+            display_list = display_list
```

Comment:
display_list.copy() for obvious thread safety?

Reply:
I thought about this and I don't think it's right. The display list, after all, is now a tree structure, so while `.copy()` looks like it might be thread safe it's actually not. Moreover, we actually don't expect the main thread to ever access the display list after making it (in a real browser that might not be true—perhaps this needs a footnote) so better to explain that logic directly. (A better implementation would avoid assigning the display list to a variable at all, but I think that would require a refactor which doesn't seem worth it.)

Reply:
Good point!